### ISSUE TITLE
UI crashes when the expected output amount of a swap is zero

### ISSUE THREAD

**Issue by atiselsts** [Uniswap/interface#6992]
**Bug Description**
I discovered that the Uniswap UI crashes with "Division by zero" error when trying to swap very small quantities of tokens that would result in zero output (due to the fact that 100% of the input is taken as a fee).

**Steps to Reproduce**

1. Go to https://app.uniswap.org/#/swap
2.  Select USDC to ETH swap
3. Enter 0.000001 in the input amount field.

**Expected Behavior**
Show output amount and price impact instead of crashing.

**Additional Context**

Error reported by the UI:

`[DecimalError] Division by zero at https://app.uniswap.org/static/js/4415.cbc0fe1d.js:2:1030173 at b.dividedBy.b.div (https://app.uniswap.org/static/js/4415.cbc0fe1d.js:2:1025229) at t.toSignificant (https://app.uniswap.org/static/js/2518.a23f0ea6.js:2:156227) at n.toSignificant (https://app.uniswap.org/static/js/2518.a23f0ea6.js:2:160338) at V (https://app.uniswap.org/static/js/2518.a23f0ea6.js:2:59921) at ft (https://app.uniswap.org/static/js/main.50fde327.js:1:1525222) at Sa (https://app.uniswap.org/static/js/638.2c9c3c94.js:2:930376) at Ss (https://app.uniswap.org/static/js/638.2c9c3c94.js:2:987489) at bc (https://app.uniswap.org/static/js/638.2c9c3c94.js:2:976595) at gc (https://app.uniswap.org/static/js/638.2c9c3c94.js:2:976523) at mc (https://app.uniswap.org/static/js/638.2c9c3c94.js:2:976386) at ac (https://app.uniswap.org/static/js/638.2c9c3c94.js:2:973166) at cc (https://app.uniswap.org/static/js/638.2c9c3c94.js:2:973557) at Hi (https://app.uniswap.org/static/js/638.2c9c3c94.js:2:914097) at https://app.uniswap.org/static/js/638.2c9c3c94.js:2:971099
`

**Comment by atiselsts** [comment]
There may be another related issue even when it does not crash. Check the price of 1ETH in the image. 1ETH is 3,747.43 USDC, is this as it's supposed to be?
![image](https://github.com/Uniswap/interface/assets/3797089/65f7f06b-f089-44d4-a708-b4bba4e7f1b4)

**Comment by lyzergic** [comment]
Can I take this?

**Comment by ItIsOHM** [comment]
Hey @lyzergic if you're working on this, can you also tell me what changes you're making? I'm a beginner here and interested in learning.

**Comment by lyzergic** [comment]
I tried recreating it locally, the day after I commented on the issue, but couldn't. In prod I did manage to recreate it on that day. 
Now a couple of days later I can't do it in neither environments. I do have a doubt that it has something to do with `SwapDetailsDropdown` component, since it was throwing once it finished loading, but I hadn't had much time that day.
I'll keep trying to re-create at different times.
